src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -4176,6 +4176,46 @@ def _check_zip_deployment_status(cmd, rg_name, name, deployment_status_url, slot
                        is still on-going. Navigate to your scm site to check the deployment status""")
     return res_dict
 
+def _get_latest_deployment_id(cmd, rg_name, name, deployment_status_url, slot):
+    import requests
+    from azure.cli.core.util import should_disable_connection_verify
+
+    headers = get_scm_site_headers(cmd.cli_ctx, name, rg_name, slot)
+    total_trials = 30
+    num_trials = 0
+    while num_trials < total_trials:
+        time.sleep(2)
+        response = requests.get(deployment_status_url, headers=headers,
+                                verify=not should_disable_connection_verify())
+        try:
+            res_dict = response.json()
+        except json.decoder.JSONDecodeError:
+            logger.warning("Deployment status endpoint %s returned malformed data. Retrying...", deployment_status_url)
+        finally:
+            num_trials = num_trials + 1
+        if 'id' in res_dict and 'temp' not in res_dict['id']:
+            return res_dict['id']
+    return None
+
+def _track_deployment_using_deploymentstatus_api(cmd, deploymentstatusapi_url, timeout=None):
+    total_trials = (int(timeout) // 2) if timeout else 450
+    num_trials = 0
+    runtime_successful = False
+    while num_trials < total_trials:
+        time.sleep(2)
+        response_body = send_raw_request(cmd.cli_ctx, "GET", deploymentstatusapi_url, log_request_response=False)
+        deployment_status = response_body.json().get('properties').get('status')
+        logger.info(deployment_status + "...")
+        if deployment_status == "RuntimeSuccessful":
+            runtime_successful = True
+            break
+        elif deployment_status == "RuntimeFailed":
+            raise CLIError("Deployment Failed: Runtime failed to start, check the deployment logs for more info")
+        elif deployment_status == "BuildFailed":
+            raise CLIError("Deployment Failed: Build failed, check the deployment logs for more info")
+    if not runtime_successful:
+        raise CLIError("Timeout reached while tracking deployment status, however, the deployment operation is still on-going. Navigate to your scm site to check the deployment status")
```

Comment:
I think we want to give more useful insights,in failure scenarios, like the logs endpoint, no. of instances updated, failed etc from the response, instead of throwing generic error. no?

Reply:
Logging the additional info